File: java/programming/Esercizi a casa Java/Potenza.java
import java.util.*;

public class Potenza{
  public static void main(String[] args){
      Scanner in=new Scanner(System.in);
      System.out.print("Inserisci una base: ");
      int base=in.nextInt();
      System.out.print("Inserisci l'esponente: ");
      int exp=in.nextInt();
      System.out.println("Risultato: "+potenza(base,exp));
  }

  public static int potenza(int base, int exp){
    if(exp==1){
      return base;
    }else{
      return base*potenza(base, exp-1);
    }
  }
}

File: c/algorithms and data structures/L03/potenza.c
#include <stdio.h>

int potenza(int b, int e){
	int ris=1,cont=0;
	if(e==0)
		return 1;
	do{
		ris=ris*b;
		cont++;
	}while(cont<e);
	return ris;
}

int main(void){
	int b,e,ris;
	printf("Inserisci la base e l'esponente: ");
	scanf("%d,%d",&b,&e);
	ris=potenza(b,e);
	printf("Risultato: %d",ris);
	return 0;
}